### ISSUE TITLE
Delete CommonRuntimeTypes

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtimelab#520]
src\coreclr\nativeaot uses a `CommonRuntimeTypes` class to access commonly used `typeof` objects. Caches like this are not great. This one also has a negative size impact since accessing any member will trigger the class constructor that will do all the `typeof` operations to populate the cache.

```csharp
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

class CommonRuntimeTypes
{
    public static Type Array { get; } = typeof(Array);
    public static Type Int32 { get; } = typeof(Int32);
    public static Type Double { get; } = typeof(Double);
    public static Type Single { get; } = typeof(Single);
    public static Type Void { get; } = typeof(void);
}

class test
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static Type Benchmark1() => typeof(void);

    [MethodImpl(MethodImplOptions.NoInlining)]
    public static Type Benchmark2() => CommonRuntimeTypes.Void;

    public static void Main()
    {
        var sw = Stopwatch.StartNew();
        for (int i = 0; i < 1000000; i++)
        {
            Benchmark2();
        }
        Console.WriteLine(sw.ElapsedMilliseconds);
        sw = Stopwatch.StartNew();
        for (int i = 0; i < 1000000; i++)
        {
            Benchmark2();
        }
        Console.WriteLine(sw.ElapsedMilliseconds);
        sw = Stopwatch.StartNew();
        for (int i = 0; i < 1000000; i++)
        {
            Benchmark2();
        }
        Console.WriteLine(sw.ElapsedMilliseconds);

        sw = Stopwatch.StartNew();
        for (int i = 0; i < 1000000; i++)
        {
            Benchmark1();
        }
        Console.WriteLine(sw.ElapsedMilliseconds);
        sw = Stopwatch.StartNew();
        for (int i = 0; i < 1000000; i++)
        {
            Benchmark1();
        }
        Console.WriteLine(sw.ElapsedMilliseconds);
        sw = Stopwatch.StartNew();
        for (int i = 0; i < 1000000; i++)
        {
            Benchmark1();
        }
        Console.WriteLine(sw.ElapsedMilliseconds);
    }
}
```

On CoreCLR, this all prints ~3.

On NativeAOT, this prints ~8 for no static static cache case and ~2 for the static cache (interestingly, accessing statics seems to be faster than CoreCLR).

If we really really care, we can make direct `typeof` even faster, but I don't think the places that use `CommonRuntimeTypes` are _that_ perf sensitive.

**Comment by jkotas** [comment]
I agree that this is a good idea.

**Comment by MichalStrehovsky** [comment]
> I agree that this is a good idea.

Cool! Marked this as a good first issue.

Basically we want to delete CommonRuntimeTypes and fix all places that use it to use `typeof` directly.

**Comment by ThomasGoulet73** [comment]
If nobody started this, I'd like to give it a shot as it seems pretty straightforward. If I understood correctly, we replace each call to CommonRuntimeTypes properties with `typeof([Type cached by the property])` and then delete CommonRuntimeTypes.cs.

**Comment by MichalStrehovsky** [comment]
@ThomasGoulet73 yes, that's the scope of this! It's yours, thank you!

**Comment by jkotas** [comment]
Fixed by #616
